lib/Doctrine/Common/Cache/CouchbaseBucketCache.php

```diff
@@ -0,0 +1,241 @@
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch($id)
+    {
+        $id = $this->normalizeKey($id);
+
+        try {
+            $document = $this->bucket->get($id);
+        } catch (Exception $e) {
+            // TODO Log exception.
+            return false;
+        }
+
+        if ($document instanceof Document && $document->value !== false) {
+            return $this->decode($document->value);
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doContains($id)
+    {
+        $id = $this->normalizeKey($id);
+
+        try {
+            $document = $this->bucket->get($id);
+        } catch (Exception $e) {
+            // TODO Log exception.
+            return false;
+        }
+
+        if ($document instanceof Document) {
+            return !$document->error;
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doSave($id, $data, $lifeTime = 0)
+    {
+        $id = $this->normalizeKey($id);
+
+        if ($lifeTime > 30 * 24 * 3600) {
+            $lifeTime = time() + $lifeTime;
+        }
+
+        try {
+            $encoded = $this->encode($data);
+
+            $document = $this->bucket->upsert($id, $encoded, [
+                'expiry' => (int) $lifeTime,
+            ]);
+        } catch (Exception $e) {
+            // TODO Log exception.
+            return false;
+        }
+
+        if ($document instanceof Document) {
+            return !$document->error;
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doDelete($id)
+    {
+        $id = $this->normalizeKey($id);
+
+        try {
+            $document = $this->bucket->remove($id);
+        } catch (Exception $e) {
+            if ($e->getCode() === 13) {
+                return true;
+            }
+
+            // TODO Log exception.
+            return false;
+        }
+
+        if ($document instanceof Document) {
+            return !$document->error;
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFlush()
+    {
+        $manager = $this->bucket->manager();
+
+        // Flush does not return with success or failure, and must be enabled per bucket on the server.
+        // Store a marker item so that we will know if it was successful.
+        $this->doSave(__METHOD__, true, 60);
+
+        $manager->flush();
+
+        if ($this->doContains(__METHOD__)) {
+            $this->doDelete(__METHOD__);
+
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doGetStats()
+    {
+        $manager = $this->bucket->manager();
+        $stats   = $manager->info();
+        $nodes = $stats['nodes'];
+        $node = $nodes[0]; // TODO Support more than 1 node.
```

Comment:
could be. should we expect that everyone will have this setup? should we document the limitation? we only get accurate stats then if they use a reverse proxy. if we don't assume one is used, could we agree upon some aggregation logic here, e.g. simply adding the values? maybe it's not worth it. you tell me : )